### PR TITLE
Make Shiny Awesome 1st draft

### PR DESCRIPTION
**Pull request by KrzysztofWrobel**

### CODE REVIEW

File: _posts_knitr/2016-12-27-make-shiny-awesome.Rmd [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+During event Marek told about details and history of package development. This simple library not only solves mentioned navigation issues, but also encourages to write more structured and better code by writing UI modules.
+
+Last, but not least Marek presented our solution for real-time collaboration for Shiny. Although Shiny does support this kind of reactivity, it is limited in terms of usage scale. 
+In a regular Shiny app you may use global scope reactive values that are shared between multiple session and refreshed everytime someone makes a change. But this kind of design can support maybe few hundreds of simultanous users, but sometime you may need like 10 thousands.
```
Comment (filipstachura) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: Is it? And how is this state persisted in this case?

File: _posts_knitr/2016-12-27-make-shiny-awesome.Rmd [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+During event Marek told about details and history of package development. This simple library not only solves mentioned navigation issues, but also encourages to write more structured and better code by writing UI modules.
+
+Last, but not least Marek presented our solution for real-time collaboration for Shiny. Although Shiny does support this kind of reactivity, it is limited in terms of usage scale. 
+In a regular Shiny app you may use global scope reactive values that are shared between multiple session and refreshed everytime someone makes a change. But this kind of design can support maybe few hundreds of simultanous users, but sometime you may need like 10 thousands. 
+In order to solve this we have created a wrapper for Reactive real-time databases (eg. RethinkDB, Firebase), which unfortunately is not published yet, but we are working on that. It allows us to span multiple Shiny apps processes with one real-time synced reactive database. So to satisfy requirement of 10 thousands users we just have to run a 100 Shiny apps processes with load balancer in the front.
```
Comment (filipstachura) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: `, which unfortunately is not published yet, but we are working on that` - wywalic

Comment (filipstachura) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: So to satisfy requirement of 10 thousands users we just have to run a 100 Shiny apps processes with load balancer in the front. - How did you get those numbers?

File: _posts_knitr/2016-12-27-make-shiny-awesome.Rmd (hunk 1 above)
Comment (KrzysztofWrobel) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: Is what I'm writing about global reactive value not true? State is not persisteted but that's not what important if you need real-time collaboration, is it?

File: _posts_knitr/2016-12-27-make-shiny-awesome.Rmd (hunk 2 above)
Comment (KrzysztofWrobel) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: Okey, makes sense

Comment (KrzysztofWrobel) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: I just assumed those to show what from my point of view is the power of reactive database. I can rewritte it in a way it doesn't use any specific numbers, is it ok?

**Comment by KrzysztofWrobel** [comment]
@filipstachura I need some anserwers to completely fix this article:)

File: _posts_knitr/2016-12-27-make-shiny-awesome.Rmd (hunk 1 above)
Comment (filipstachura) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: Okey, I've checked it is true. I think that two drawbacks of this approach are:
- state is not persisted
- this data cannot be used outside of Shiny app (what if you want to analyse the content in those reactive values?)
- scalability

File: _posts_knitr/2016-12-27-make-shiny-awesome.Rmd (hunk 2 above)
Comment (filipstachura) [inline _posts_knitr/2016-12-27-make-shiny-awesome.Rmd]: having numbers is good but we need to make sure of how many concurrent users can be served by one server

File: _posts/2016-12-27-make-shiny-awesome.md [hunk 3]
```diff
@@ -17,26 +17,31 @@ A week before Christmas our CTO, Marek Rogala gave a speech about ways to make S
...
+Last, but not least Marek presented our **real-time collaboration solution** for Shiny. Although Shiny does support this kind of reactivity, it is limited in many ways. 
+In a regular Shiny app you may use global scope reactive values that are shared between multiple session and refreshed everytime someone makes a change. There are some limitiations though:
 
-After Marek presentation we have moved to local pub for pizza and beer, where we had a really great time talking about R, Shiny and totally data science stuff with a local R enthusiast community.
+- state is not persisted, unless stored and initialised from database
+- this data cannot be used outside of Shiny app - what if you want to analyse the content in those reactive values?
+- scalability - apps hosted on **shinyapp.io** can handle [few hundreds users](https://www.rstudio.com/products/shinyapps-io-performance-tuning/) and Shiny Server Pro instance will support up to [200 users depnding on a chosen license](https://support.rstudio.com/hc/en-us/articles/220760228))
```
Comment (filipstachura) [inline _posts/2016-12-27-make-shiny-awesome.md]: typo: depnding

**Pull request merged**